projects/packages/forms/src/modules/form/view.js

```diff
@@ -13,6 +13,26 @@ import {
 import { validateField } from '../../contact-form/js/validate-helper';
 import { focusNextInput, dispatchSubmitEvent, submitForm } from './shared';
 
+const isEmptyValue = value => {
+	if ( value === null || value === undefined ) {
+		return true;
+	}
+
+	if ( typeof value === 'string' && value.trim() === '' ) {
+		return true;
+	}
+
+	if ( Array.isArray( value ) && value.length === 0 ) {
+		return true;
```

Comment:
I think we can use recursion here, like `return value.length === 0 || value.every( isEmptyValue )`